src/common/chains.ts

```diff
@@ -100,98 +133,139 @@ export const ChainNames: Record<ChainIndices, string> = {
     UX: 'UX Network',
 }
 
+@Struct.type('telos_account_voter_info')
+export class TelosAccountVoterInfo extends API.v1.AccountVoterInfo {
+    @Struct.field(Int64) last_stake!: Int64
+}
+
+@Struct.type('telos_account_object')
+export class TelosAccountObject extends API.v1.AccountObject {
+    @Struct.field(TelosAccountVoterInfo, {optional: true})
+    declare voter_info?: TelosAccountVoterInfo
+}
+
+@Struct.type('wax_account_voter_info')
+export class WAXAccountVoterInfo extends API.v1.AccountVoterInfo {
+    @Struct.field(Float64) declare unpaid_voteshare: Float64
+    @Struct.field(TimePoint) declare unpaid_voteshare_last_updated: TimePoint
+    @Struct.field(Float64) declare unpaid_voteshare_change_rate: Float64
+    @Struct.field(TimePoint) declare last_claim_time: TimePoint
+}
+
+@Struct.type('wax_account_object')
+export class WAXAccountObject extends API.v1.AccountObject {
+    @Struct.field(WAXAccountVoterInfo, {optional: true}) declare voter_info?: WAXAccountVoterInfo
+}
+
 /**
  * An exported list of ChainDefinition entries for select chains.
  */
-export const Chains: Record<ChainIndices, ChainDefinition> = {
-    EOS: ChainDefinition.from({
+export namespace Chains {
```

Comment:
For some reason, the generics don't work at all if we don't use a namespace here.